.github/workflows/dotnetcore.yml

```diff
@@ -19,11 +19,15 @@ jobs:
     - name: Setup .NET Core
       uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: 6.0.x
+        dotnet-version: 7.0.x
     - name: Build with dotnet
       run: dotnet build --configuration Release
     - name: Test with dotnet test
       run: dotnet test --configuration Release --no-build
+    - name: Nerdbank.GitVersioning
+      uses: dotnet/nbgv@v0.4.0
+      id: nbgv
```

Comment:
I have just realised that this option gives a lot of deprecation warnings because it uses the `set-output` command.
I would suggest doing _one_ of these:
- fixing dotnet/nbgv to not use `set-output`
- adding a step that does the same thing (install nbgv and save the version as an output)
- revert to what we were doing before

Reply:
(A little bit of experimentation and failure later) I've now merge some additional changes to fix these. This resolves both the build warnings - I'm happy to explicitly accept that we still want net5.0 support for now and for the changes to cope the sensible new security policy which has been enacted which means we move to the environment variable model for nerdbank.

Reply:
And now fixed the extra double quote. But stuck with ${{ env.NAME }} syntax because I couldn't get ${NAME} to work.